doc/use_cases/turtlebot3_tutorial.md

```diff
@@ -0,0 +1,239 @@
+
+This tutorial may take about 2-5 hours to complete completely depends on your experience with ROS, robots and what computer system you have.
+
+## Requirements:
+
+- [Install ROS2]([https://index.ros.org/doc/ros2/Installation/](https://index.ros.org/doc/ros2/Installation/))
+
+- Install Navigation2
+
+    - `sudo apt install ros-<ros2-distro>-navigation2`
+
+    - `sudo apt install ros-<ros2-distro>-ros-dashing-nav2-bringup`
+
+- [Build Turtlebot3](http://emanual.robotis.com/docs/en/platform/turtlebot3/ros2/#setup)
+
+- [Install Gazebo](http://gazebosim.org/tutorials?tut=install_ubuntu&cat=install)
+
+## Navigation2 with Turtlebot 3 in Gazebo
+
+### 0- Edit .bashrc
+
+First, make sure you have the following lines in your .bashrc file.
+
+Open a new terminal and open your .bashrc file which is located in your home directory. You can use your favorite text editor.
+
+```
+sudo gedit ~/.bashrc
+```
+
+Add the following lines to the end of the file
+
+```
+source /opt/ros/<ros2_distro>/setup.bash
+source ~/turtlebot3_ws/install/setup.bash
+export TURTLEBOT3_MODEL=waffle
+export GAZEBO_MODEL_PATH=$GAZEBO_MODEL_PATH:~/turtlebot3_ws/src/turtlebot3/turtlebot3_simulations/turtlebot3_gazebo/models
+```
+
+save the file and close it. Whenever we open a new terminal, this script will be executed. 
+
+*Please note that this will  make your terminal's start-up time considerably longer, so when you are done with this tutorial you may want to remove these lines from your .bashrc file.*
+
+### 1- Launch Gazebo
+
+Now, launch Gazebo with the world model, open a new terminal and type
+
+    gazebo --verbose -s libgazebo_ros_init.so /home/<user_name>/turtlebot3_ws/src/turtlebot3/turtlebot3_simulations/turtlebot3_gazebo/worlds/turtlebot3_worlds/waffle.model
+
+Don't forget to change <user_name> to your actual user name. Once, Gazebo is launched, you should see the Turtlebot3 world and Turtlebot 3.
+
+If Gazebo fails to start, run the following commands and try to launch Gazebo again.
+
+    killall gzserver
+    killall gzclient
+
+![enter image description here](https://github.com/mlherd/ros2/blob/master/img/gazebo_turlebot1.PNG?raw=true)
+
+### 2- Launch Turtlebot 3 Robot State Publisher
+ 
+Launch Turtlebot 3 specific nodes,
+
+    ros2 launch turtlebot3_bringup turtlebot3_state_publisher.launch.py use_sim_time:=True
+
+Make sure use_sim_time is set to **True**.
+
+### 3- Launch Navigation2 
+
+Launch Navigation 2. If you set `autostart:=False`, you need to click on the start button in RVIZ to initialize the nodes. Make sure `use_sim time` is set to **True**, because we want to use the time simulation time in Gazebo instead of the system time.
+
+turtlebot3_world.yaml is the configuration file for the map we want to provide Navigation2. In this case, it has the map resolution value, threshold values for obstacles and free spaces, and a map file location.
+
+```
+ros2 launch nav2_bringup nav2_bringup_launch.py use_sim_time:=True autostart:=False \
```

Comment:
We have a turtlebot3 simulation launch file with bringup, gazebo, and rviz. Why not let them use that vs all this manual stuff?

Reply:
I wanted to show the user a step by step tutorial, since this knowledge can be useful when the debug issues and use nav2 with different robots.

Reply:
Good call, I'd add a note at the start saying theres a single launch file that will do this for you after doing this tutorial to understand

Reply:
I added an additional step called shortcut to the tutorial :)